datalad/tests/test_nonasyncrunner.py

```diff
@@ -0,0 +1,159 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil; coding: utf-8 -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Test the thread based runner (aka. non asyncio based runner).
+"""
+import asyncio
+import os
+import queue
+import signal
+import subprocess
+from time import sleep, time
+
+from datalad.tests.utils import assert_true, eq_, known_failure_windows, \
+    known_failure_osx, with_tempfile
+
+from ..cmd import WitlessProtocol, WitlessRunner, StdOutCapture
+from ..nonasyncrunner import ReaderThread, run_command
+
+
+@known_failure_windows
```

Comment:
could you add a comment on what is so special about windows here?

Reply:
> sorry that my suggestion for [bd7bb59](https://github.com/datalad/datalad/commit/bd7bb5929eb04e68483d95f19e06c7e7a085ee6f) was not entire kosher . it is what broke CI and running tests locally. let me know if you need a hand to mitigate

Don't worry, its fine. It's a small fix. I combine it with the logging improvements. :-)

Reply:
added